### PR TITLE
Update redis-rs to 0.12 in bb8-redis dependencies.

### PR DESCRIPTION
**Pull request by singulared**

### CODE REVIEW

File: redis/Cargo.toml [hunk 1]
```diff
@@ -10,4 +10,4 @@ repository = "https://github.com/khuey/bb8"
 bb8 = { path = "../" }
 futures = "0.1"
 tokio = "0.1"
-redis = "0.11"
+redis = "0.12"
```
Comment (khuey) [inline redis/Cargo.toml]: This requires bumping the version of bb8-redis as well to 0.4.0-pre, since the new redis is semver incompatible.

Comment (singulared) [reply redis/Cargo.toml]: Yep, I think you are right. Is this a problem?

Comment (khuey) [reply redis/Cargo.toml]: No, it's not a problem.

**Comment by khuey** [comment]
Thanks!

**Pull request merged**